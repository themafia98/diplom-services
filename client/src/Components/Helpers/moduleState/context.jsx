import { createContext } from 'react';

const ModuleContextState = createContext();
const ModuleContextActions = createContext();

export { ModuleContextState, ModuleContextActions };
