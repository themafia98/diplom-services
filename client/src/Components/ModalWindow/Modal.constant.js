const ACTIONS = {
  CHANGE_TYPE: 'CHANGE_TYPE',
  CHANGE_DESCRIPTION: 'CHANGE_DESCRIPTION',
  CHANGE_ROOT: 'CHANGE_ROOT',
};

export { ACTIONS };
