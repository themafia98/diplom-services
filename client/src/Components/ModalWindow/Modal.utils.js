import moment from 'moment';
import { ACTIONS } from './Modal.constant';

const modalState = {
  visible: false,
  reg: {
    name: null,
    password: null,
    departament: null,
    email: null,
    surname: null,
  },
  jurnal: {
    timeLost: null,
    date: moment().format('DD.MM.YYYY HH:mm:ss'),
    description: null,
  },
  description: {
    value: null,
  },
  error: new Set(),
  loading: false,
  taskStatus: null,
  type: null,
};

const reducer = (state, { type, payload }) => {
  switch (type) {
    case ACTIONS.CHANGE_TYPE:
      return {
        ...state,
        type: payload,
      };
    case ACTIONS.CHANGE_ROOT:
      return {
        ...state,
        ...payload,
      };
    case ACTIONS.CHANGE_DESCRIPTION:
      return {
        ...state,
        description: payload,
      };
    default:
      return state;
  }
};

export { reducer, modalState };
