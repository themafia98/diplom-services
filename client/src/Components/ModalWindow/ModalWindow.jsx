import React from 'react';
import { modalWindowType } from './types';
import _ from 'lodash';
import moment from 'moment';
import { Modal, Button, message, Select } from 'antd';
import { TASK_CONTROLL_JURNAL_SCHEMA } from 'Models/Schema/const';
import { createNotification, isTimeLostValue } from 'Utils';
import SimpleEditableModal from './SimpleEditableModal';
import RegistrationModal from './RegistrationModal';
import ActionList from 'Components/ActionList';
import TrackerModal from './TrackerModal';
import MailResponserModal from './MailResponserModal';

import Textarea from 'Components/Textarea';
import modelContext from 'Models/context';

const { Option } = Select;

class ModalWindow extends React.PureComponent {
  state = {
    visible: false,
    reg: {
      name: null,
      password: null,
      departament: null,
      email: null,
      surname: null,
    },
    jurnal: {
      timeLost: null,
      date: moment().format('DD.MM.YYYY HH:mm:ss'),
      description: null,
    },
    description: {
      value: this.props.editableContent || null,
    },
    error: new Set(),
    loading: false,
    taskStatus: null,
    type: null,
  };

  static contextType = modelContext;
  static propTypes = modalWindowType;
  static defaultProps = {
    onEdit: null,
    modeControll: null,
    onRejectEdit: null,
    modeEditContent: null,
    defaultView: false,
    onCancelEditModeContent: null,
    onUpdateEditable: null,
    rulesEdit: true,
    rulesStatus: true,
    isLoadList: true,
    routeDataActive: {},
    actionTypeList: 'default',
    mode: '',
    actionType: '',
    keyTask: '',
    udata: {},
    path: '',
    statusTaskValue: '',
    accessStatus: [],
  };

  static getDerivedStateFromProps = (props, state) => {
    const { mode } = props;
    const { type = '' } = state;

    if (mode === 'reg' && _.isNull(type)) {
      return {
        ...state,
        type: 'regType',
      };
    }
    return state;
  };

  onMessage = (event) => {
    message.warning('Вы в режиме редактирования карточки.');
  };

  onChangeDescription = (event) => {
    const { currentTarget: { value: valueTarget = '' } = {} } = event;
    const { description = {} } = this.state;

    if (valueTarget || valueTarget === '')
      this.setState({
        ...this.state,
        description: {
          ...description,
          value: valueTarget,
        },
      });
  };

  showModal = (event, type = '') => {
    const { [type]: visibleType = false } = this.state;

    this.setState({
      ...this.state,
      [type]: !visibleType,
      visible: true,
      type,
    });
  };

  onRegUser = async () => {
    const { reg: { name, password, departament, email, surname } = {}, type = '', loading } = this.state;
    const { Request } = this.context;

    if (!name || !password || !departament || !email || loading) {
      message.warning('Не все поля заполнены');
      return;
    }

    try {
      const rest = new Request();
      const res = await rest.sendRequest(
        '/reg',
        'POST',
        {
          email,
          password,
          displayName: `${name} ${surname}`,
          departament,
          position: 'Master',
          rules: 'full',
          accept: true,
        },
        false,
      );

      if (!res || res.status !== 200) {
        throw new Error('Bad registration data');
      }

      this.setState({ ...this.state, [type]: false });
    } catch (error) {
      if (error?.response?.status !== 404) console.error(error);
    }
  };

  onSaveEdit = (event) => {
    const { description: { value: valueDescription = '' } = {}, type: typeState = '' } = this.state;

    const {
      onUpdate,
      routeDataActive = {},
      routeDataActive: { key = null } = {},
      onCancelEditModeContent,
    } = this.props;
    onUpdate({
      key,
      updateBy: 'key',
      id: routeDataActive?._id,
      updateItem: valueDescription,
      updateField: 'description',
      item: { ...routeDataActive },
      store: 'tasks',
    })
      .then((res) => {
        if (onCancelEditModeContent) onCancelEditModeContent(event);
        this.setState({
          ...this.state,
          [typeState]: false,
          type: null,
          loading: false,
        });
        message.success('Описание изменено.');
      })
      .catch((error) => {
        message.error('Ошибка редактирования.');
      });
  };

  getTemplate = (nameTask, timeLost, description, date) => {
    return `
    Работа над задачей ${nameTask}.\n
    Время затрачено: ${timeLost}.\n
    Дата: ${date}. \n
    Описание: ${description}. \n
    `;
  };

  onTrackTask = async () => {
    const { jurnal, type = '' } = this.state;

    const {
      onCaching,
      routeDataActive: { key = null, name: nameTask = '' } = {},
      actionType,
      keyTask,
      udata: { displayName = '', _id: uid = '' } = {},
    } = this.props;
    const item = { ...jurnal, depKey: keyTask, editor: displayName, uid };
    const { timeLost = '', description = '', date = '' } = jurnal || {};

    if (onCaching) {
      await onCaching({ item, actionType, depStore: 'tasks', store: 'jurnalworks' });
      this.handleCancel();
    }
    const itemNotification = {
      type: 'global',
      title: 'Списание времени в журнал',
      isRead: false,
      message: this.getTemplate(nameTask, timeLost, description, date),
      action: {
        type: 'tasks_link',
        moduleName: 'taskModule',
        link: key,
      },
      uidCreater: uid,
      authorName: displayName,
    };

    createNotification('global', itemNotification).catch((error) => {
      if (error?.response?.status !== 404) console.error(error);
      message.error('Ошибка глобального уведомления');
    });

    return this.setState({
      ...this.state,
      [type]: false,
      type: null,
      loading: false,
      jurnal: { timeLost: null, date: moment().format('DD.MM.YYYY HH:mm:ss'), description: null },
      error: new Set(),
    });
  };

  onChangeStatusTask = async (customStatus = null) => {
    const { taskStatus = null, type = '' } = this.state;

    const { onUpdate, routeDataActive = {}, routeDataActive: { key = null } = {}, path } = this.props;

    if (_.isNull(taskStatus) && !_.isString(customStatus))
      return this.setState({
        ...this.state,
        [type]: false,
        type: null,
        loading: false,
      });

    try {
      await onUpdate({
        path,
        id: routeDataActive?._id,
        key,
        updateBy: 'key',
        updateItem: _.isString(customStatus) ? customStatus : taskStatus,
        updateField: 'status',
        item: { ...routeDataActive },
        store: 'tasks',
      });

      this.setState({
        ...this.state,
        [type]: false,
        type: null,
        loading: false,
      });
      message.success('Статус изменен.');
    } catch (error) {
      message.error('Ошибка редактирования.');
    }
  };

  handleOk = async (event, action = '', response = '') => {
    const { type } = this.state;
    debugger;
    if (action === 'close') {
      this.showModal(event, type, response, action);
      this.onChangeStatusTask('Закрыт');
    }

    if (type === 'regType') return this.onRegUser(event);
    else if (type === 'jur' && this.validation()) return this.onTrackTask();
    else if (type === 'statusTask') return this.onChangeStatusTask();
  };

  handleCancel = (event) => {
    const { type = '' } = this.state;
    this.setState({
      ...this.state,
      [type]: false,
      loading: false,
      jurnal: { timeLost: null, date: moment().format('DD.MM.YYYY HH:mm:ss'), description: null },
      error: new Set(),
    });
  };

  onChangeSelect = (event) => {
    const { type, reg = {} } = this.state;
    if (type === 'statusTask') {
      this.setState({ ...this.state, taskStatus: event });
    } else this.setState({ ...this.state, reg: { ...reg, departament: event } });
  };

  onChange = (event) => {
    const { type = '' } = this.state;
    const { target: { value = '', className = '' } = {} } = event;
    const param = className?.split(' ')[1];
    if (!param) return;

    const newState =
      type === 'regType'
        ? {
            ...this.state,
            reg: {
              ...this.state.reg,
              [param]: value,
            },
          }
        : {
            ...this.state,
            [param]: value,
          };

    this.setState(newState);
  };

  validation = () => {
    const {
      jurnal: { timeLost = null, date = moment(), description = null },
      error = [],
      udata: { displayName = '' } = {},
      type = '',
    } = this.state;

    const { keyTask, modeEditContent } = this.props;
    const { schema = {} } = this.context;
    let _valid = true;
    let invalidDate = !date || !_.isDate(new Date(date));
    let invalidTimeLost = !timeLost || !_.isString(timeLost) || !isTimeLostValue(timeLost);
    let invalidDescription = !description || !_.isString(description);

    if (type === 'jur' && !modeEditContent && (invalidDate || invalidTimeLost || invalidDescription)) {
      _valid = false;
      const errorBundle = error.size ? new Set([...error]) : new Set();
      message.error('Не все поля заполнены!');

      if (invalidDate) errorBundle.add('date');
      else if (errorBundle.has('date')) errorBundle.delete('date');
      if (invalidTimeLost) errorBundle.add('timeLost');
      else if (errorBundle.has('timeLost')) errorBundle.delete('timeLost');
      if (invalidDescription) errorBundle.add('description');
      else if (errorBundle.has('description')) errorBundle.delete('description');
      this.setState({ ...this.state, error: errorBundle });
    }
    if (!_valid) return _valid;

    const validData = schema?.getSchema(TASK_CONTROLL_JURNAL_SCHEMA, {
      depKey: keyTask,
      timeLost: timeLost,
      editor: displayName,
      date: date,
      description: description,
    });

    if (validData) return _valid;
    else return false;
  };

  onChangeTask = (event) => {
    if (!event) return;
    const { jurnal = {} } = this.state;
    const { target: { value = '', className = '' } = {}, _isValid = null } = event;
    const typeChanges = className?.split(' ')[1];

    if (value && typeChanges === 'timeLost') {
      this.setState({
        ...this.state,
        jurnal: { ...jurnal, timeLost: value },
      });
    } else if (_isValid && event && _isValid) {
      this.setState({
        ...this.state,
        jurnal: { ...jurnal, date: event.toString() },
      });
    } else if ((value || value === '') && typeChanges === 'description') {
      this.setState({
        ...this.state,
        jurnal: { ...jurnal, description: value },
      });
    }
  };

  renderRegistrationModal = () => {
    const { type = '' } = this.state;
    const { [type]: visible = false } = this.state;
    return (
      <>
        <Button aria-label="reg-button" type="primary" onClick={(event) => this.showModal(event, 'regType')}>
          Регистрация
        </Button>
        <Modal
          className="modalWindow"
          visible={visible}
          onOk={this.handleOk}
          onCancel={this.handleCancel}
          title={'Регистрация'}
        >
          <RegistrationModal cbOnChange={this.onChange} cbOnChangeSelect={this.onChangeSelect} />
        </Modal>
      </>
    );
  };

  renderChangerStatusModal = (actionProps = {}) => {
    const { statusTaskValue, accessStatus } = this.props;
    const { type = '' } = this.state;
    const { [type]: visible = false } = this.state;
    return (
      <div className="dropDownWrapper">
        <ActionList {...actionProps} />
        <Modal
          className="modalWindow changeStatus"
          visible={visible}
          onOk={this.handleOk}
          onCancel={this.handleCancel}
          title="Смена статуса"
        >
          <Select onChange={this.onChangeSelect} defaultValue={statusTaskValue}>
            {accessStatus.map((status, i) =>
              i === 0 ? (
                <Option key={i + status} value={statusTaskValue}>
                  {statusTaskValue}
                </Option>
              ) : (
                <Option key={i + status} value={status}>
                  {status}
                </Option>
              ),
            )}
          </Select>
        </Modal>
      </div>
    );
  };

  render() {
    const {
      onEdit,
      modeControll,
      onRejectEdit,
      modeEditContent,
      defaultView,
      onCancelEditModeContent,
      onUpdateEditable,
      rulesEdit,
      rulesStatus,
      isLoadList,
      routeDataActive,
      actionTypeList: viewType = 'default',
    } = this.props;
    const { type: typeView = '' } = this.state;
    const {
      [typeView]: visible = false,
      description: { value: valueDesc = '' } = {},
      error,
      jurnal: { description, timeLost },
    } = this.state;

    if (defaultView) {
      return <SimpleEditableModal {...this.props} />;
    }

    if (typeView === 'regType') {
      return this.renderRegistrationModal();
    }

    const actionProps = {
      viewType,
      entityName: 'taskView',
      onMessage: this.onMessage,
      showModal: this.showModal,
      modeControll,
      onUpdateEditable: onUpdateEditable,
      onRejectEdit: onRejectEdit,
      isLoadList,
      rulesStatus,
      rulesEdit,
      onEdit,
    };

    if (typeView === 'statusTask') {
      return this.renderChangerStatusModal({
        ...actionProps,
      });
    }

    const trackProps = {
      typeView,
      visible,
      handleOk: this.handleOk,
      handleCancel: this.handleCancel,
      onChangeTask: this.onChangeTask,
      error,
      timeLost,
      description,
    };
    const isVisibleMailResponser = visible && typeView === 'mailResponse';

    return (
      <>
        <div className="dropDownWrapper">
          <ActionList {...actionProps} showModal={this.showModal} />
        </div>
        {isVisibleMailResponser ? (
          <MailResponserModal
            handleCancel={this.handleCancel}
            handleOk={this.handleOk}
            routeDataActive={routeDataActive}
            typeView={typeView}
            visibleModal={isVisibleMailResponser}
          />
        ) : (
          <Modal
            className="modalWindow"
            visible={visible || modeEditContent}
            onOk={this.handleOk}
            destroyOnClose={true}
            onCancel={modeEditContent ? onCancelEditModeContent : this.handleCancel}
            title={modeEditContent ? 'Редактирование' : 'Отчет времени'}
          >
            {modeEditContent ? (
              <Textarea
                key="textAreaEdit"
                className="editContentDescription"
                //editor={true}
                row={10}
                onChange={this.onChangeDescription}
                value={valueDesc ? valueDesc : ''}
                defaultValue={valueDesc ? valueDesc : ''}
              />
            ) : visible && typeView === 'jur' ? (
              <TrackerModal {...trackProps} />
            ) : null}
          </Modal>
        )}
      </>
    );
  }
}

export default ModalWindow;
