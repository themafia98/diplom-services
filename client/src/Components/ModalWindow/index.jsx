import React from "react";
import { Modal, Button } from "antd";
import uuid from "uuid/v4";
import RegistrationModal from "./RegistrationModal";
class ModalWindow extends React.Component {
    state = {
        login: null,
        visible: false,
        name: null,
        password: null,
        departament: null,
        email: null,
        loading: false,
    };

    showModal = () => {
        this.setState({
            visible: true,
        });
    };

    handleOk = e => {
        const { login, name, password, departament, email, loading, surname } = this.state;
        const { firebase } = this.props;

        if (login && name && password && departament && email && !loading) {
            firebase
                .registration(email, password)
                .then(res => {
                    if (res.additionalUserInfo.isNewUser)
                        firebase.db.collection("users").add({
                            uuid: uuid(),
                            login: login,
                            name: name,
                            surname: surname,
                            departament: departament,
                            email: email,
                            rules: "false",
                            position: "Не установлено",
                            status: "Новый сотрудник",
                        });
                })
                .then(res => {
                    this.setState({
                        uuid: uuid(),
                        visible: false,
                        loading: false,
                    });
                })
                .catch(error => console.error(error));
        }
    };

    handleCancel = event => {
        this.setState({
            visible: false,
            loading: false,
        });
    };

    onChangeSelect = event => {
        this.setState({ departament: event });
    };

    onChange = event => {
        const { target } = event;
        if (target.className.split(" ")[1] === "surname") {
            this.setState({
                surname: target.value,
            });
        } else if (target.className.split(" ")[1] === "name") {
            this.setState({
                name: target.value,
            });
        } else if (target.className.split(" ")[1] === "password") {
            this.setState({
                password: target.value,
            });
        } else if (target.className.split(" ")[1] === "login") {
            this.setState({
                login: target.value,
            });
        } else if (target.className.split(" ")[1] === "email") {
            this.setState({
                email: target.value,
            });
        }
    };

    render() {
        const { mode } = this.props;
        if (mode !== "reg") return null;
        else
            return (
                <React.Fragment>
                    {mode === "reg" ? (
                        <Button type="primary" onClick={this.showModal}>
                            Регистрация
                        </Button>
                    ) : null}
                    <Modal
                        className="modalWindow"
                        visible={this.state.visible}
                        onOk={this.handleOk}
                        onCancel={this.handleCancel}
                        title={mode === "reg" ? "Регистрация" : null}
                    >
                        {mode === "reg" ? (
                            <RegistrationModal cbOnChange={this.onChange} cbOnChangeSelect={this.onChangeSelect} />
                        ) : (
                            <div></div>
                        )}
                    </Modal>
                </React.Fragment>
            );
    }
}
export default ModalWindow;
