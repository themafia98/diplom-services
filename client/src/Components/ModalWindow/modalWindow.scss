.modalWindow {
  input {
    margin: 5px;
  }

  .ant-calendar-picker {
    width: 100%;
  }

  .ant-calendar-time-picker-btn {
    display: none;
  }

  .ant-modal-body {
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  button {
    margin-top: 5px;
  }

  .selectDepartament {
    width: 100%;
  }

  .errorFild {
    border: 1px solid red;
  }
}

.modal-content {
  display: flex;
  flex-direction: column;
  padding: 20px;
  input,
  .ant-select {
    margin: 5px 0;
  }
}

.dropDownWrapper {
  .ant-spin {
    margin-left: 5px;
  }
}
