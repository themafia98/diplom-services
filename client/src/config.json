{
    "title": "Controll system",
    "icon": "/img/logo.svg",
    "firebase": {
        "session": "LOCAL",
        "lowConnection": "false"
    },
    "clientDB": {
        "name": "controllSystemDB",
        "version": 1
    },
    "forceUpdate": false,
    "msTimeoutLoading": 0,
    "tabsLimit": 15,
    "menu": [
        {
            "EUID": "mainModule",
            "VALUE": "Главное меню",
            "ICON": "home",
            "SIGN": "default",
            "PARENT_CODE": null
        },
        {
            "EUID": "cabinetModule",
            "VALUE": "Личный кабинет",
            "ICON": "user",
            "PARENT_CODE": null
        },
        {
            "EUID": "taskModule",
            "VALUE": "Задачи",
            "ICON": "file-done",
            "PARENT_CODE": null
        },
        {
            "EUID": "taskModule_all",
            "VALUE": "Все задачи",
            "ICON": null,
            "PARENT_CODE": "taskModule"
        },
        {
            "EUID": "taskModule_myTasks",
            "VALUE": "Мои задачи",
            "ICON": null,
            "PARENT_CODE": "taskModule"
        },
        {
            "EUID": "taskModule_сalendar",
            "VALUE": "Календарь задач",
            "ICON": null,
            "PARENT_CODE": "taskModule"
        },
        {
            "EUID": "taskModule_createTask",
            "VALUE": "Новая задача",
            "ICON": null,
            "SIGN": "freedom"
        },
        {
            "EUID": "customersModule",
            "VALUE": "Работа с клиентами",
            "ICON": "global",
            "PARENT_CODE": null
        },
        {
            "EUID": "customersModule_contacts",
            "VALUE": "Контакты",
            "ICON": null,
            "PARENT_CODE": "customersModule"
        },
        {
            "EUID": "customersModule_reception",
            "VALUE": "Обращения",
            "ICON": null,
            "PARENT_CODE": "customersModule"
        },
        {
            "EUID": "contactModule",
            "VALUE": "Связь",
            "ICON": "wechat",
            "PARENT_CODE": null
        },
        {
            "EUID": "contactModule_feedback",
            "VALUE": "Информация",
            "ICON": null,
            "PARENT_CODE": "contactModule"
        },
        {
            "EUID": "contactModule_chat",
            "VALUE": "Корпоративный чат",
            "ICON": null,
            "PARENT_CODE": "contactModule"
        },
        {
            "EUID": "statisticModule",
            "VALUE": "Статистика",
            "ICON": "line-chart",
            "PARENT_CODE": null
        },
        {
            "EUID": "settingsModule",
            "VALUE": "Настройки",
            "ICON": "build",
            "PARENT_CODE": null
        }
    ]
}
