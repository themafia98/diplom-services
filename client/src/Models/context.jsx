import { createContext } from 'react';
import config from '../config.json';
import Request from '../Models/Rest';
import Schema from '../Models/Schema';
import Notification from '../Models/NotificationRouter';
import { clientDB } from './ClientSideDatabase';

export const modelMethods = {
  clientDB,
  rest: new Request(),
  Request,
  config,
  schemaModel: Schema,
  schema: new Schema('no-strict'),
  Notification,
};

const modelsContext = createContext(modelMethods);

export default modelsContext;
