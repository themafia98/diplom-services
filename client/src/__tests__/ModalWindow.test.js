import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import config from '../config.json';
import ModalWindow from '../Components/ModalWindow';
import context from '../Models/context';
describe('<ModalWindow />', () => {
  test('Should works with mode', () => {
    const props = {
      onCaching: () => {},
      primaryKey: '',
      routeDataActive: {},
      mode: 'reg',
      path: '',
      typeRequst: '',
      keyTask: '',
      accessStatus: ['Открыт', 'Выполнен', 'Закрыт', 'В работе'],
      onUpdate: () => {},
      onEdit: () => {},
      onRejectEdit: () => {},
      modeControll: '',
      editableContent: '',
      modeEditContent: false,
      onCancelEditModeContent: () => {},
      onUpdateEditable: () => {},
      statusTaskValue: 'В работе',
    };
    const ModalWindowWrapper = mount(<ModalWindow {...props} />, { context });

    expect(toJson(ModalWindowWrapper)).toMatchSnapshot();
    expect(ModalWindowWrapper.find('Button').text()).toEqual('Регистрация');

    ModalWindowWrapper.find('Button').forEach((node) => {
      expect(node.simulate('click', [])).toBeTruthy();
    });

    const nextProps = {
      ...props,
      mode: '',
    };

    const ModalWindowWrapperNoMode = shallow(<ModalWindow {...nextProps} />);
    expect(toJson(ModalWindowWrapperNoMode)).toMatchSnapshot();
    expect(ModalWindowWrapperNoMode).toEqual({});

    const nextPropsMode = {
      ...props,
      mode: 'jur',
      typeState: 'statusTask',
    };

    const ModalWindowWrapperJurStatus = shallow(<ModalWindow {...nextPropsMode} />);
    expect(toJson(ModalWindowWrapperJurStatus)).toMatchSnapshot();

    const nextPropsModeEdit = {
      ...props,
      mode: 'jur',
      typeState: '',
      modeControll: 'edit',
      modeEditContent: true,
    };

    const ModalWindowWrapperJurEditMode = shallow(<ModalWindow {...nextPropsModeEdit} />);
    expect(toJson(ModalWindowWrapperJurEditMode)).toMatchSnapshot();

    const nextPropsModeEditFalse = {
      ...props,
      mode: 'jur',
      typeState: '',
      modeControll: 'edit',
      modeEditContent: false,
    };

    const ModalWindowWrapperJurEditModeFalse = shallow(<ModalWindow {...nextPropsModeEditFalse} />);
    expect(toJson(ModalWindowWrapperJurEditModeFalse)).toMatchSnapshot();

    const nextPropsModeEditBags = {
      ...props,
      mode: 'jur',
      typeState: '',
      modeControll: 'default',
      modeEditContent: true,
    };

    const ModalWindowWrapperJurEditModeBags = shallow(<ModalWindow {...nextPropsModeEditBags} />);
    expect(toJson(ModalWindowWrapperJurEditModeBags)).toMatchSnapshot();
  });
});
