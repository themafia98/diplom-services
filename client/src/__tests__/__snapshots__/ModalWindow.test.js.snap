// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalWindow /> Should works with mode 1`] = `
<ModalWindow
  accessStatus={
    Array [
      "Открыт",
      "Выполнен",
      "Закрыт",
      "В работе",
    ]
  }
  editableContent=""
  firebase={Object {}}
  keyTask=""
  mode="reg"
  modeControll=""
  modeEditContent={false}
  onCaching={[Function]}
  onCancelEditModeContent={[Function]}
  onEdit={[Function]}
  onRejectEdit={[Function]}
  onUpdate={[Function]}
  onUpdateEditable={[Function]}
  path=""
  primaryKey=""
  routeDataActive={Object {}}
  statusTaskValue="В работе"
  typeRequst=""
>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    type="primary"
  >
    <Wave>
      <button
        className="ant-btn ant-btn-primary"
        onClick={[Function]}
        type="button"
      >
        <span
          key=".0"
        >
          Регистрация
        </span>
      </button>
    </Wave>
  </Button>
  <Modal
    className="modalWindow"
    confirmLoading={false}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    title="Регистрация"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <Component
      className="modalWindow"
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <Icon
            className="ant-modal-close-icon"
            type="close"
          />
        </span>
      }
      confirmLoading={false}
      footer={
        <LocaleReceiver
          componentName="Modal"
          defaultLocale={
            Object {
              "cancelText": "Cancel",
              "justOkText": "OK",
              "okText": "OK",
            }
          }
        >
          [Function]
        </LocaleReceiver>
      }
      maskTransitionName="fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      onOk={[Function]}
      prefixCls="ant-modal"
      title="Регистрация"
      transitionName="zoom"
      visible={false}
      width={520}
      wrapClassName=""
    >
      <PortalWrapper
        visible={false}
      />
    </Component>
  </Modal>
</ModalWindow>
`;

exports[`<ModalWindow /> Should works with mode 2`] = `""`;

exports[`<ModalWindow /> Should works with mode 3`] = `
<Fragment>
  <div
    className="dropDownWrapper"
  >
    <Dropdown
      mouseEnterDelay={0.15}
      mouseLeaveDelay={0.1}
      overlay={
        <React.Fragment>
          <Menu>
            <MenuItem>
              <p
                className="jur"
                onClick={[Function]}
              >
                Занести в журнал работы
              </p>
            </MenuItem>
            <MenuItem>
              <p
                className="statusTask"
                onClick={[Function]}
              >
                Сменить статус задачи
              </p>
            </MenuItem>
            <MenuItem>
              <p
                className="statusTask"
                onClick={[Function]}
              >
                Редактировать задачу
              </p>
            </MenuItem>
          </Menu>
        </React.Fragment>
      }
      placement="bottomLeft"
    >
      <p>
        Управление задачей
        <Icon
          type="down"
        />
      </p>
    </Dropdown>
  </div>
  <Modal
    confirmLoading={false}
    maskTransitionName="fade"
    okType="primary"
    transitionName="zoom"
    visible={false}
    width={520}
  />
</Fragment>
`;

exports[`<ModalWindow /> Should works with mode 4`] = `
<Fragment>
  <div
    className="dropDownWrapper"
  >
    <Dropdown
      mouseEnterDelay={0.15}
      mouseLeaveDelay={0.1}
      overlay={
        <React.Fragment>
          <Menu>
            <MenuItem>
              <p
                className="jur"
                onClick={[Function]}
              >
                Занести в журнал работы
              </p>
            </MenuItem>
            <MenuItem>
              <p
                className="statusTask"
                onClick={[Function]}
              >
                Сменить статус задачи
              </p>
            </MenuItem>
            <MenuItem>
              <p
                className="statusTask"
                onClick={[Function]}
              >
                Редактировать задачу
              </p>
            </MenuItem>
          </Menu>
        </React.Fragment>
      }
      placement="bottomLeft"
    >
      <p>
        Управление задачей
        <Icon
          type="down"
        />
      </p>
    </Dropdown>
    <p
      className="modeControllEdit"
      onClick={[Function]}
    >
      Сохранить изменения
    </p>
    <p
      className="modeControllEditReject"
      onClick={[Function]}
    >
      Отмена изменений
    </p>
  </div>
  <Modal
    className="modalWindow"
    confirmLoading={false}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    title="Редактирование"
    transitionName="zoom"
    visible={true}
    width={520}
  >
    <Textarea
      className="editContentDescription"
      defaultValue=""
      key="textAreaEdit"
      onChange={[Function]}
      row={10}
      value=""
    />
  </Modal>
</Fragment>
`;

exports[`<ModalWindow /> Should works with mode 5`] = `
<Fragment>
  <div
    className="dropDownWrapper"
  >
    <Dropdown
      mouseEnterDelay={0.15}
      mouseLeaveDelay={0.1}
      overlay={
        <React.Fragment>
          <Menu>
            <MenuItem>
              <p
                className="jur"
                onClick={[Function]}
              >
                Занести в журнал работы
              </p>
            </MenuItem>
            <MenuItem>
              <p
                className="statusTask"
                onClick={[Function]}
              >
                Сменить статус задачи
              </p>
            </MenuItem>
            <MenuItem>
              <p
                className="statusTask"
                onClick={[Function]}
              >
                Редактировать задачу
              </p>
            </MenuItem>
          </Menu>
        </React.Fragment>
      }
      placement="bottomLeft"
    >
      <p>
        Управление задачей
        <Icon
          type="down"
        />
      </p>
    </Dropdown>
    <p
      className="modeControllEdit"
      onClick={[Function]}
    >
      Сохранить изменения
    </p>
    <p
      className="modeControllEditReject"
      onClick={[Function]}
    >
      Отмена изменений
    </p>
  </div>
  <Modal
    confirmLoading={false}
    maskTransitionName="fade"
    okType="primary"
    transitionName="zoom"
    visible={false}
    width={520}
  />
</Fragment>
`;

exports[`<ModalWindow /> Should works with mode 6`] = `
<Fragment>
  <div
    className="dropDownWrapper"
  >
    <Dropdown
      mouseEnterDelay={0.15}
      mouseLeaveDelay={0.1}
      overlay={
        <React.Fragment>
          <Menu>
            <MenuItem>
              <p
                className="jur"
                onClick={[Function]}
              >
                Занести в журнал работы
              </p>
            </MenuItem>
            <MenuItem>
              <p
                className="statusTask"
                onClick={[Function]}
              >
                Сменить статус задачи
              </p>
            </MenuItem>
            <MenuItem>
              <p
                className="statusTask"
                onClick={[Function]}
              >
                Редактировать задачу
              </p>
            </MenuItem>
          </Menu>
        </React.Fragment>
      }
      placement="bottomLeft"
    >
      <p>
        Управление задачей
        <Icon
          type="down"
        />
      </p>
    </Dropdown>
  </div>
  <Modal
    className="modalWindow"
    confirmLoading={false}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    title="Редактирование"
    transitionName="zoom"
    visible={true}
    width={520}
  >
    <Textarea
      className="editContentDescription"
      defaultValue=""
      key="textAreaEdit"
      onChange={[Function]}
      row={10}
      value=""
    />
  </Modal>
</Fragment>
`;
